sentry-kotlin-multiplatform/src/appleMain/kotlin/io/sentry/kotlin/multiplatform/SentryScope.kt

```diff
@@ -0,0 +1,164 @@
+package io.sentry.kotlin.multiplatform
+
+import io.sentry.kotlin.multiplatform.extensions.toCocoaBreadcrumb
+import io.sentry.kotlin.multiplatform.extensions.toCocoaSentryLevel
+import io.sentry.kotlin.multiplatform.extensions.toCocoaSentryUser
+import platform.Foundation.NSDictionary
+import platform.Foundation.allKeys
+import cocoapods.Sentry.SentryScope as CocoaScope
+
+actual class SentryScope : ISentryScope {
+
+    // We are directly modifying the Cocoa SDK scope
+    private var scope: CocoaScope? = null
+
+    actual override var level: SentryLevel?
+        set(value) {
+            scope?.setLevel(value!!.toCocoaSentryLevel())
+        }
+        get() {
+            val levelMap = scope?.serialize()?.get("level")
+            if (levelMap != null) {
+                val levelString = (levelMap as String).uppercase()
+                return SentryLevel.valueOf(levelString)
+            }
+            return null
+        }
+
+    actual override var user: SentryUser?
+        set(value) {
+            value?.onFieldChanged = {
+                scope?.setUser(value?.toCocoaSentryUser())
+            }
+        }
+        get() {
+            val userMap = scope?.serialize()?.get("user") as Map<String, String>?
+            if (userMap != null) {
+                val sentryUser = SentryUser()
+                sentryUser.email = userMap["email"] as String
+                sentryUser.username = userMap["username"] as String
+                sentryUser.id = userMap["id"] as String
+                sentryUser.ipAddress = userMap["ip_address"] as String
+                return sentryUser
```

Comment:
Those conversions from Map to Object could be extension methods like `SentryUser.fromMap(map): SentryUser`